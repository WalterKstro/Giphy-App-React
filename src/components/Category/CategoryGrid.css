img{
    width: 100%;
    display: block;
    height:auto;
}
.category_item{
    font-size: 2rem;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
    grid-template-rows: repeat(5, auto);
    gap: 2rem;
    grid-auto-flow: dense;
}

.category_title{
    background-color: #20232a;
    color:white;
    padding:1rem 2rem;
    grid-row: 1 / 2;
    grid-column: 1 / -1;
}

.loading{
    font-size: 10rem;
    text-align: center
}