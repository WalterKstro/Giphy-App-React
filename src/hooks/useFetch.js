import { useEffect, useState } from "react"
import getGiphyCategories from '../helpers/getGiphyCategories'

const useFetch = (category) => {
    const [state, setState] = useState({
        data : [],
        loading : true
    })

    useEffect(() => {
        getGiphyCategories(category)
        .then( response => {
            setState({
                data : response,
                loading : false
            })
        } )
    },[category])

    return state
}


export default useFetch