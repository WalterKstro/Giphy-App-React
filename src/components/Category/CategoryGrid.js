import useFetch from '../../hooks/useFetch'
import './CategoryGrid.css'
import CategoryItem from './CategoryItem'
import PropTypes from 'prop-types'



const GridCategory = ({ category }) => {

    
    const {data, loading} = useFetch(category)
    
    return (
        <div className="category_item">
            <p className="category_title">Category: <strong>{ category }</strong></p>
            {loading && <p className="loadding">Loadding data ...</p>}
            {
                data.map( item => (
                    <CategoryItem 
                        key={item.id} 
                        {...item}
                    />
                ))
            }
        </div>
    )
}

GridCategory.propTypes = {
    category : PropTypes.string.isRequired
}

export default GridCategory
