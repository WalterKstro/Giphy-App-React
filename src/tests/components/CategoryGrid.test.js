import { shallow } from "enzyme"
import CategoryGrid from '../../components/Category/CategoryGrid'
import useFetch from "../../hooks/useFetch"
jest.mock('../../hooks/useFetch')

describe('Testing in the component <CategoryGrid/>', () => {
    const category = 'Naruto'


    test('should show a snaptshot of component <CategoryGrid/>', () => {
        useFetch.mockReturnValue({
            data : [],
            loading : true
        })

        const component = shallow( <CategoryGrid category={category}/> )
        expect( component).toMatchSnapshot()
    })


    test('should show item with the data', () => {
        const gifts = [{id:'_se56qs23',url: 'www.facebook.com',title:'Naruto'}]

        useFetch.mockReturnValue({
            data : gifts,
            loading : false
        })
        const component = shallow( <CategoryGrid category={category}/> )
        expect( component).toMatchSnapshot()
    })
    

})
