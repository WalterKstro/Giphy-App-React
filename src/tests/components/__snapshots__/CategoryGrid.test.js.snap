// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in the component <CategoryGrid/> should show a snaptshot of component <CategoryGrid/> 1`] = `
<div
  className="category_item"
>
  <p
    className="category_title"
  >
    Category: 
    <strong>
      Naruto
    </strong>
  </p>
  <p
    className="loadding"
  >
    Loadding data ...
  </p>
</div>
`;

exports[`Testing in the component <CategoryGrid/> should show item with the data 1`] = `
<div
  className="category_item"
>
  <p
    className="category_title"
  >
    Category: 
    <strong>
      Naruto
    </strong>
  </p>
  <CategoryItem
    id="_se56qs23"
    key="_se56qs23"
    title="Naruto"
    url="www.facebook.com"
  />
</div>
`;
